### PR TITLE
Add tests for new upload/download requests

### LINKED ISSUE #803: Add tests for new upload/download requests
We need more tests!

Related to https://github.com/nspcc-dev/neofs-rest-gw/pull/206

### PR DESCRIPTION
**Pull request by tatiana-nspcc**
Close #803.

### CODE REVIEW

File: pytest_tests/lib/helpers/http_gate.py [hunk 1]
```diff
@@ -23,8 +23,8 @@
 ASSETS_DIR = os.getenv("ASSETS_DIR", "TemporaryDir/")
 
 
-@allure.step("Get via HTTP Gate")
-def get_via_http_gate(
+@allure.step("Get via REST Gate")
```
Comment (evgeniiz321) [inline pytest_tests/lib/helpers/http_gate.py:26]: Maybe rename this file itself and the dir `http_gw` as well

**Comment by evgeniiz321** [comment]
Do we need to update the commit hash - https://github.com/nspcc-dev/neofs-testcases/blob/master/.github/workflows/run-tests.yml#L147 ? I would like to see all rest test passed in this PR

File: pytest_tests/lib/helpers/http_gate.py (hunk 1 above)
Comment (tatiana-nspcc) [reply pytest_tests/lib/helpers/http_gate.py:26]: Yep. I've renamed this file and `http_gw/http_utils.py` as well

**Review by evgeniiz321 — APPROVED** [review]

**Comment by roman-khimov** [comment]
Conflicts.

**Comment by tatiana-nspcc** [comment]
> Conflicts.

I've rebased it.

File: pytest_tests/lib/helpers/http_gate.py [hunk 2]
```diff
@@ -46,7 +46,7 @@ def get_via_rest_gate(
     if download:
         download_attribute = "?download=true"
     if request_path is None:
-        request = f"{endpoint}/get/{cid}/{oid}{download_attribute}"
+        request = f"{endpoint}/objects/{cid}/by_id/{oid}{download_attribute}"
```
Comment (roman-khimov) [inline pytest_tests/lib/helpers/http_gate.py:49]: Do we have tests for old APIs after that?

File: pytest_tests/tests/services/rest_gate/test_rest_gate_new.py [hunk 3]
```diff
@@ -0,0 +1,524 @@
...
+            resp = get_via_rest_gate(
+                cid=cid,
+                oid=oid,
+                endpoint=gw_params["endpoint"],
+                return_response=True,
+            )
+            assert resp.headers["Content-Type"] == "CoolContentType"
+
+    @allure.title("Verify Content-Type if uploaded with multipart Content-Type")
+    def test_put_http_get_http_with_multipart_content_type(self, gw_params):
```
Comment (roman-khimov) [inline pytest_tests/tests/services/rest_gate/test_rest_gate_new.py:222]: What is so `multipart` here?

**Review by roman-khimov — COMMENTED** [review]
The last two can be squashed

File: pytest_tests/tests/services/rest_gate/test_rest_gate_new.py (hunk 3 above)
Comment (tatiana-nspcc) [reply pytest_tests/tests/services/rest_gate/test_rest_gate_new.py:222]: Nothing really, it's a copy-paste from the same function in the old upload, and there wasn't any multipart either. I'd better rename both of them.

File: pytest_tests/lib/helpers/http_gate.py (hunk 2 above)
Comment (tatiana-nspcc) [reply pytest_tests/lib/helpers/http_gate.py:49]: We have tests for the old upload function, but not for `/get*`. I know an easy way to add some tests for object download, and I'll take care of that.

Comment (roman-khimov) [reply pytest_tests/lib/helpers/http_gate.py:49]: OK, I want old APIs to remain being tested, they'll be kept for a long time, so we shouldn't throw away tests that we have for them.

File: pytest_tests/tests/services/rest_gate/test_rest_gate_new.py (hunk 3 above)
Comment (tatiana-nspcc) [reply pytest_tests/tests/services/rest_gate/test_rest_gate_new.py]: Done.

File: pytest_tests/lib/helpers/http_gate.py (hunk 2 above)
Comment (tatiana-nspcc) [reply pytest_tests/lib/helpers/http_gate.py:49]: Done.

File: .github/workflows/run-tests.yml [hunk 4]
```diff
@@ -144,7 +144,7 @@ jobs:
         uses: actions/checkout@v4
         with:
           repository: nspcc-dev/neofs-rest-gw
-          ref: f024e798a5bfc5d3183bac417c8a6713e9e756b0
+          ref: 5750187cbc01b06de8bb026c67bb312fa9f99c63
```
Comment (roman-khimov) [inline .github/workflows/run-tests.yml:147]: Can be v0.9.0 now.

**Review by roman-khimov — APPROVED** [review]

**Review by evgeniiz321 — APPROVED** [review]

**Pull request merged**